File: modul 3/src/kegiatan2/Kegiatan2.py
queue = []

while True:
    print("=== Antrian Pembelian Eskrim Mixue ===")
    print("1. Tambah Antrian")
    print("2. Proses Pembelian")
    print("3. Jumlah Antrian")
    print("4. Eskrim yang akan diproses selanjutnya")
    print("5. Tampilkan Antrian")
    print("6. Keluar")

    choice = input("Masukkan pilihan: ")

    if choice == "1":
        eskrim_name = input("Masukkan nama eskrim: ")
        queue.append(eskrim_name)
        print("Eskrim", eskrim_name, "telah ditambahkan ke dalam antrian")

    elif choice == "2":
        if not queue:
            print("Antrian kosong")
        else:
            print("Eskrim", queue.pop(0), "telah diproses")

    elif choice == "3":
        print("Jumlah antrian saat ini:", len(queue))

    elif choice == "4":
        if not queue:
            print("Tidak ada eskrim")
        else:
            print("Eskrim yang akan diproses selanjutnya:", queue[0])

    elif choice == "5":
        if not queue:
            print("Antrian kosong")
        else:
            print("Semua Antrian:")
            for i, eskrim in enumerate(queue, 1):
                print(i, ".", eskrim)

    elif choice == "6":
        print("Keluar dari sistem")
        break

    else:
        print("Pilihan tidak valid!")

    print()

File: modul 3/src/kegiatan2/Kegiatan2.java
package kegiatan2;

import java.util.LinkedList;
import java.util.Scanner;

public class Kegiatan2 {

    private static class Node {
        String data;
        Node next;

        Node(String data) {
            this.data = data;
        }
    }

    private static class Queue {
        Node first, last;

        void enqueue(String data) {
            Node nodeBaru = new Node(data);
            if (first == null) {
                first = nodeBaru;
                last = nodeBaru;
            } else {
                last.next = nodeBaru;
                last = nodeBaru;
            }
        }

        void dequeue() {
            if (first == null) {
                System.out.println("Antrian kosong");
            } else {
                System.out.println("Pelanggan " + first.data + " telah membeli eskrim Mixue");
                first = first.next;
            }
        }

        String peek() {
            if (first == null) {
                return "Tidak ada pelanggan";
            } else {
                return first.data;
            }
        }

        boolean isEmpty() {
            return first == null;
        }

        int size() {
            int size = 0;
            Node current = first;
            while (current != null) {
                size++;
                current = current.next;
            }
            return size;
        }

        void display() {
            Node current = first;
            int i = 1;
            while (current != null) {
                System.out.println(i + ". " + current.data);
                current = current.next;
                i++;
            }
        }
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        Queue eskrimQueue = new Queue();

        boolean session = true;
        int choice;
        String pelanggan;

        while (session) {
            System.out.println("=== Antrian Beli Eskrim Mixue ===");
            System.out.println("1. Tambah Antrian");
            System.out.println("2. Proses Pembelian");
            System.out.println("3. Jumlah Antrian");
            System.out.println("4. Pelanggan yang akan dilayani selanjutnya");
            System.out.println("5. Tampilkan Antrian");
            System.out.println("6. Keluar");

            System.out.print("Masukkan pilihan: ");
            choice = Integer.parseInt(input.nextLine());

            switch (choice) {
                case 1:
                    System.out.print("Masukkan nama pelanggan: ");
                    pelanggan = input.nextLine();
                    eskrimQueue.enqueue(pelanggan);
                    System.out.println("Pelanggan " + pelanggan + " telah ditambahkan ke dalam antrian");
                    break;
                case 2:
                    eskrimQueue.dequeue();
                    break;
                case 3:
                    System.out.println("Jumlah antrian saat ini: " + eskrimQueue.size());
                    break;
                case 4:
                    System.out.println("Pelanggan yang akan dilayani selanjutnya: " + eskrimQueue.peek());
                    break;
                case 5:
                    System.out.println("Semua Antrian:");
                    eskrimQueue.display();
                    break;
                case 6:
                    if (eskrimQueue.isEmpty()) {
                        System.out.println("Telah keluar dari sistem");
                        session = false;
                    } else {
                        System.out.println("Masih ada pelanggan yang belum dilayani!");
                    }
                    break;
                default:
                    System.out.println("Pilihan tidak valid!");
                    break;
            }
            System.out.println();
        }
    }
}
